games/space-invaders/Space Invaders [David Winter].json

```diff
@@ -0,0 +1,108 @@
+{
+  "title": "Space Invaders",
+  "authors": [
+    {
+      "id": "david-winter",
+      "name": "David Winter"
+    }
+  ],
+  "release": "1978",
+  "category": "GAME",
+  "description": "The well known game. Destroy the invaders with your ship.",
+  "images": [
+    "./Space Invaders [David Winter] - main screen.png",
+    "./Space Invaders [David Winter] - game play.png"
+  ],
+  "interpreter": {
+    "id": "chip8-david-winter",
+    "name": "CHIP8 by David Winter"
+  },
+  "binaries": {
+    "names": [
+      { "value": "Space Invaders [David Winter] - 0.9.1" },
+      { "value": "Space Invaders [David Winter]" },
+
+      { "value": "Space Invaders [David Winter] - 0.9.0" },
+      { "value": "Space Invaders [David Winter] (alt)" },
+      { "value": "INVADERS" }
+    ],
+    "extensions": [
+      "c8",
+      "ch8",
+      ""
+    ],
+    "hashes": [
+      {
+        "type": "SHA256",
+        "value": "00bb7001de52b562d98357d33375406c6d0308466d7d95c2580537d2451a75cd"
+      },
+      {
+        "type": "SHA256",
+        "value": "2d0e1fa53216b297e74041d4fb766f42327a42893e83bb4ec931a9dff5c2dd10"
+      }
+    ]
+  },
+  "sources": {
+    "names": [
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.0"
+      },
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.1"
+      }
+    ],
+    "extensions": [
+      "8o",
+      "asm",
+      "txt",
+      ""
+    ]
+  },
+  "platform": {
+    "name": "chip8",
+    "startAddress": "0x200",
+    "resolution": {
+      "width": 64,
+      "height": 32
+    }
+  },
```

Comment:
Platform became an object.

`startAddress` adds support for ROMs for ETI-660 or custom ones (David Winter's Boot-128 starts at `0x100`).

Resolution is part of the platform. I guess there are maybe some games for `schip` which don't use hires?

Reply:
Yes, SCHIP games doesn't necessarily need to use `hires`. Apart from that I can't think of any examples where resolution varies within a platform.

One thing to think about is "hybrid" games, ie. games that use machine code subroutines (`0NNN`). Should they be their own "platform", since they need to run on an actual emulated CPU?

What about games that technically aren't CHIP-8, since they require a modified interpreter to run? They mostly need to be loaded at address `0x000` and also need to run on emulated hardware, but for completion's sake I'd like to include them. (A mild example of this is the "hires" CHIP-8 games, not the SCHIP ones but the 64x64 games using two pages of display buffer: https://chip-8.github.io/extensions/#two-page-display-for-chip-8

Reply:
> What about games that technically aren't CHIP-8, since they require a modified interpreter to run?

We could reuse some of the official names as platform names and come up with some new platform names for unnamed modifications? Consistent naming scheme would be helpful here.

> One thing to think about is "hybrid" games, ie. games that use machine code subroutines (0NNN). Should they be their own "platform", since they need to run on an actual emulated CPU?

I guess we could list the MLSes here with opcode and description and emulator could check if  it supports them and if not, stop and inform the user (the default is to ignore unknown `0x0MMM` instructions) In case of gallery the emulator could show such information before even launching / fetching the app. This would also be a great documentation for developers.

Reply:
So I added `variant` field which could describe what kind of `chip8` is required. It is a subtype. Options are now part of the platform. hybrid games can include `nativeRoutines` section. I included a sample. If MLS is `required` it means the emulator can't just ignore it.